src/ck-core/cklocation.h

```diff
@@ -444,20 +494,22 @@ class CkLocMgr : public IrrGroup
   {
     return CMK_RANK_0(map->procNum(mapHandle, idx));
   }
-  inline bool isHome(const CkArrayIndex& idx) const
+  inline bool isHome(const CkArrayIndex& idx) const { return homePe(idx) == CkMyPe(); }
+  inline int whichPE(const CmiUInt8 id) const { return cache->whichPE(id); }
+  inline int whichPE(const CkArrayIndex& idx) const
   {
-    return (bool)(homePe(idx) == CkMyPe());
+    CmiUInt8 id;
+    if (!lookupID(idx, id)) return -1;
+    return cache->whichPE(id);
   }
-  int whichPE(const CkArrayIndex& idx) const;
-  int whichPE(const CmiUInt8 id) const;
-  /// Return the "last-known" location (returns a processor number)
-  int lastKnown(const CkArrayIndex& idx);
-  int lastKnown(CmiUInt8 id);
-
-  inline void insertID(const CkArrayIndex& idx, const CmiUInt8 id)
+  inline int lastKnown(const CmiUInt8 id) const { return cache->lastKnown(id); }
+  inline int lastKnown(const CkArrayIndex& idx) const
   {
-    if (compressor) return;
-    idx2id[idx] = id;
+    int pe = whichPE(idx);
```

Comment:
Why doesn't this use `cache->lastKnown(id)` a la `whichPE`?

Reply:
For now I left it as is because homePe always has to convert from id to index, so since we already have the index, in the common case I wanted to avoid going from index to ID to index. Some of this will have to change anyways once we completely remove CkLocMgr from CkLocCache.